jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/BoundaryEventWithOutputSetPersistenceTest.java

```diff
@@ -0,0 +1,153 @@
+        KieSession ksession = runtimeEngine.getKieSession();
+        long ksessionId = ksession.getIdentifier();
+        assertTrue("session id not saved.", ksessionId > 0);
+        
+        HumanTaskMockHandler humanTaskMockHandler = new HumanTaskMockHandler();
+        ProcessInstance process = registerHTHandlerAndStartProcess(ksession, humanTaskMockHandler);
+        
+        sleepAndVerifyTimerRuns(process.getState());
+        List<? extends VariableInstanceLog> logs = getLogService().findVariableInstances(process.getId());
+        Assert.assertEquals(3, logs.size());
+    
+        // The process completes
+        process = ksession.getProcessInstance(process.getId());
+        assertNull("Expected process to have been completed and removed", process);
+    }
+
+    private ProcessInstance registerHTHandlerAndStartProcess(KieSession ksession, HumanTaskMockHandler humanTaskMockHandler) { 
+        // Register Human Task Handler
+        ksession.getWorkItemManager().registerWorkItemHandler(WORK_ITEM_HANLDER_TASK, humanTaskMockHandler);
+    
+        // Start the process 
+        ProcessInstance process = ksession.startProcess(PROCESS_NAME);
+        long processId = process.getId();
+        assertTrue("process id not saved", processId > 0);
+        
+        // The process is in the Human Task waiting for its completion
+        int processState = process.getState();
+        assertEquals("Expected process state to be " + processStateName[ProcessInstance.STATE_ACTIVE] + " not "
+                + processStateName[processState], ProcessInstance.STATE_ACTIVE, processState);
+        
+        return process;
+    
+    }
+
+    private void completeWork(KieSession ksession, HumanTaskMockHandler humanTaskMockHandler) {
+        assertTrue("The work item task handler does not have a work item!", humanTaskMockHandler.workItem != null);
+        long workItemId = humanTaskMockHandler.workItem.getId();
+        assertTrue("work item id not saved", workItemId > 0);
+        
+        // The Human Task is completed
+        Map<String, Object> results = new HashMap<String, Object>();
+        try {
+            ksession.getWorkItemManager().completeWorkItem(workItemId, results);
+        } catch (Exception e) {
+            logger.warn("Work item could not be completed!");
+            e.printStackTrace();
+            fail(e.getClass().getSimpleName() + " thrown when completing work item: " + e.getMessage());
+        }
+    }
+
+    private void sleepAndVerifyTimerRuns(int processState) throws InterruptedException { 
+        // wait 3 seconds to see if the boss is notified
+        if (processState == ProcessInstance.STATE_ACTIVE) {
+            int sleep = 2000;
+            logger.debug("Sleeping {} seconds", sleep / 1000);
+            Thread.sleep(sleep);
+            logger.debug("Awake!");
+        }
+       
+    }
+
+    
+
+    private static class HumanTaskMockHandler implements WorkItemHandler {
```

Comment:
This class is registered as a work item handler, but never used afterwards in the test. Shouldn't we try to recover HumanTaskNodeInstance object from here as per requirements in BAPL? wdyt?

Reply:
it is used now